test/datadog/e2e_eks_test.go

```diff
@@ -0,0 +1,79 @@
+package datadog
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/runner"
+	"github.com/DataDog/datadog-agent/test/new-e2e/utils/infra"
+	"github.com/DataDog/helm-charts/test/common"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/eks"
+	"github.com/pulumi/pulumi/sdk/v3/go/auto"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	ddaPrefix = "dda-datadog"
+	dcaPrefix = "dda-datadog-cluster-agent"
+	ccPrefix  = "dda-datadog-clusterchecks"
+)
+
+func Test_E2E_AgentOnEKS(t *testing.T) {
+	// Create pulumi EKS stack
+	config := common.SetupConfig()
+	stackConfig := runner.ConfigMap{
+		"ddinfra:aws/eks/linuxBottlerocketNodeGroup": auto.ConfigValue{Value: "false"},
+		"ddinfra:aws/eks/windowsNodeGroup":           auto.ConfigValue{Value: "false"},
+		"pulumi:disable-default-providers":           auto.ConfigValue{Value: "[]"},
+		"aws:skipCredentialsValidation":              auto.ConfigValue{Value: "true"},
+		"aws:skipMetadataApiCheck":                   auto.ConfigValue{Value: "false"},
+	}
+	stackConfig.Merge(config)
+
+	_, stackOutput, err := infra.GetStackManager().GetStack(context.Background(), "eks-e2e", stackConfig, eks.Run, false)
```

Comment:
It's not obvious to me what chart version this will install. Is it local or the latest published?

Reply:
The EKS pulumi scenario installs the [latest](https://github.com/DataDog/test-infra-definitions/blob/f18cee7730ecf0cd2c35561d79912d092257fbcf/components/datadog/agent/helm.go#L111-L117) agent helm chart version. The `test-infra-definitions` framework doesn't have functionality to pass in custom chart values, but we could add it in the future. I'll put a comment for now